src/components/order/StatusLabel/index.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useMemo } from 'react'
+import styled from 'styled-components'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faCheckCircle, faClock, faDotCircle } from '@fortawesome/free-solid-svg-icons'
+
+import { OrderStatus } from 'api/operator'
+
+export type Props = { status: OrderStatus }
+
+const Wrapper = styled.div<Props>`
+  font-size: ${({ theme }): string => theme.fontSizeNormal};
+  font-weight: ${({ theme }): string => theme.fontWeightBold};
+  text-transform: capitalize;
+
+  border-radius: 0.4rem;
+
+  padding: 0.75em;
+  display: inline-block;
+
+  & > svg {
+    margin: 0 0.75rem 0 0;
+  }
+
+  ${({ theme, status }): string => {
+    let background, color
+
+    switch (status) {
+      case 'expired':
+        color = theme.labelTextExpired
+        background = theme.labelBgExpired
+        break
+      case 'filled':
+      case 'partially filled':
+        color = theme.labelTextFilled
+        background = theme.labelBgFilled
+        break
+      case 'open':
+        color = theme.labelTextOpen
+        background = theme.labelBgOpen
+        break
+    }
+    return `
+      background: ${background};
+      color: ${color};
+    `
+  }}
+`
+
+export const StatusLabel = (props: Props): JSX.Element => {
+  const { status } = props
+
+  const icon = useMemo(() => {
```

Comment:
honestly I would just move this out of the component as a normal functional component:
```tsx
function getOrderIcon(status: OrderStatus) {
    switch (status) {
      case 'expired':
        return faClock
      case 'filled':
      case 'partially filled':
        return faCheckCircle
      case 'open':
        return faDotCircle
    }
}

const StatusIcon = ({ status }: { status: OrderStatus }) => {
  const icon = getOrderIcon(status)
  
  return <FontAwesomeIcon icon={icon} />
}

export const StatusLabel = ({ status }: Props): JSX.Element => (
    <Wrapper status={status}>
      <StatusIcon status={status} />
      {status}
    </Wrapper>
  )
```

But that's just me. I never really like building components within another

Reply:
Makes sense, will update (in a new PR)

Reply:
Follow up https://github.com/gnosis/gp-ui/pull/114

Reply:
good call 👌